atlasdb-feign/src/main/java/com/palantir/atlasdb/http/AtlasDbFeignTargetFactory.java

```diff
@@ -179,14 +180,26 @@ private AtlasDbFeignTargetFactory() {
                 type,
                 RefreshableProxyInvocationHandler.create(
                         configPollingRefreshable.getRefreshable(),
-                        serverListConfig -> createProxyWithFailover(
-                                serverListConfig.sslConfiguration().map(sslSocketFactoryCreator),
-                                serverListConfig.proxyConfiguration().map(proxySelectorCreator),
-                                serverListConfig.servers(),
-                                feignConnectTimeout,
-                                feignReadTimeout,
-                                maxBackoffMillis,
-                                type,
-                                userAgent)));
+                        serverListConfig -> {
+                            if (serverListConfig.hasAtLeastOneServer()) {
+                                return createProxyWithFailover(
```

Comment:
`<insert rant about methods with many parameters>`

No action here - we can't reasonably make it look better than this.

Reply:
Yep... This project was designed to be a very thin wrapper around Feign, so I didn't like the option of moving the logic for creating an `SSLSocketFactory` or `ProxySelector` into this project.